### PR TITLE
Site creator information

### LINKED ISSUE #31: Save site creator as a user in pivot table
**To use:** `$site->owner;`

**Instead of:** `User::find($site->user_id);`

NOTE: You need to change site migration

### PR DESCRIPTION
**Pull request by DIOHz0r**
This PR is for close #31

### CODE REVIEW

**Review by rorepoid — CHANGES_REQUESTED** [review]
Hi @DIOHz0r , I like your changes, but I was seeing too many tests are failing, and change all these tests to pass is maybe laborious.

Do you think it would be better to add an accessor like `getOwnerAttribute` instead of renaming the column `user_id` to `owner`?

In that case, I made a mistake saying migration should be changed 😅

**Comment by DIOHz0r** [comment]
I hope this works for you

File: app/Http/Livewire/Sites/ShowSite.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ public function render()
     public function mount(Site $site)
     {
         $this->site = $site;
-        $this->user = User::find($site->user_id);
+        $this->user = User::find($site->getSiteOwner());
     }
```
Comment (rorepoid) [inline app/Http/Livewire/Sites/ShowSite.php:26]: We can also do with a cleaner way with an [accessor](https://laravel.com/docs/8.x/eloquent-mutators#defining-an-accessor). That's some like this ⬇️ 
```
public function mount(Site $site)
    {
        $this->site = $site;
        $this->user = $site->owner;
    }
```

File: app/Models/Site.php [hunk 2]
```diff
@@ -15,4 +15,9 @@ public function users()
     {
         return $this->belongsToMany(User::class)->withTimestamps();
     }
+
+    public function getSiteOwner()
+    {
+        return $this->user_id;
+    }
```
Comment (rorepoid) [inline app/Models/Site.php:22]: And the Accessor(the function) should be named like this ⬇️ (to Accessor works). And then inside the function call the user to find the user with user_id of the site
```
    getOwnerAttribute()
```

**Review by rorepoid — COMMENTED** [review]
I was reading laravel documentation, and I found [Accessors](https://laravel.com/docs/8.x/eloquent-mutators#defining-an-accessor) could be helpful.

maybe we have learned something new? 😄 what do you think?

File: app/Models/Site.php (hunk 2 above)
Comment (DIOHz0r) [reply app/Models/Site.php]: I update all code according the review.

**Review by rorepoid — APPROVED** [review]
Yeah, I will approve this because this works fine

I'd like to show another way to do that

In Site model 
```
    public function getOwnerAttribute()
    {
        return User::find($this->user_id);
    }
```

In the component
```
    $this->user = $site->owner;   // alternative to $this->user = User::find($site->owner);
```

And it also works fine

**Comment by rorepoid** [comment]
Nice @DIOHz0r 
Thank you to participate! 🎉

**Pull request merged**